src/contexts/transaction.context.tsx

```diff
@@ -4,7 +4,7 @@ import { TransactionData } from "src/domain";
 
 interface TransactionContext {
   transaction?: TransactionData;
-  setTransaction: (transaction: TransactionData) => void;
+  setTransaction: (transaction?: TransactionData) => void;
```

Comment:
Why make this optional? When does it make sense to "set" an `undefined` transaction?